flybywiresim/a32nx
# feat(hyd): aerodynamic model for moving parts

Fixes #[issue_no]

## Summary of Changes


Implements an aerodynamic drag and lift model that can be applied to any moving part, be it cargo door, gear doors, control surfaces....

Any hydraulically controlled component will have to fight aerodynamic forces, providing a realistic feedback for all control surfaces.

- [x] Recreate local relative wind and add it to systems context
- [x] Basic drag model
- [x] Basic lift model
- [x] Refactoring aero forces calculations
- [x] Creating an "aero testbed" in unit tests
- [x] Rework of hydraulic controllers in edge case situations with massive aerodynamic forces
- [x] Tuning of control surface actuator


## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub):

## Testing instructions


Standard operations:
1. Check that current aileron droop function on power off works properly
2. Check that aileron control when hydraulic power is available works properly (no wobble, positioning is correct)
3. Shutting down all hydraulic power in flight should result in aileron staying close to neutral position due to relative wind

Ludicrous operations:
   Use custom weather with editable wind
1. Wind from back of the plane should accelerate droop down if aileron surface is lower than horizontal position.
2. Wind from the back of the plane should droop an aileron UP if hydraulic power is lost while aileron is above horizontal position.
3. Same as 1. & 2. with plane upside down
4. Max wind (150kts) from any direction: aileron position control should still be satisying: good positioning. No lag. No wobbling.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
